Login: improve the invalid WP.com email error screen

jaclync opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7483 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR follows the Android changes to improve the existing invalid WP.com email error screen. The main changes are:

- The "Enter Your Store Address" button is hidden when the merchant starts from the login with store address flow
- Added a "What is WordPress.com" button similar to other places on the login flow
  - The existing `ULErrorViewController` does not support additional views below the current UI. In order to make it more flexible for any other customizations, I added an outlet for the content stack view and an optional `auxiliaryView: View?` in the ULError view model. `NotWPAccountViewModel` then configures a `UIButton` for the pre-existing CTA to find the email
  - A new optional view controller parameter is passed to `ULErrorViewModel.viewDidLoad` call because the button handlers often need the view controller to present other UI. This isn't a very ideal, please feel free to suggest other ways that work with the current ULError screen technical design
  - A click event is tracked `what_is_wordpress_com_on_invalid_email_screen`, it's slightly different from Android because this screen is inside the WPAuthenticator and click events are embedded in `unified_login_interaction`
- [Copy changes ](https://github.com/woocommerce/woocommerce-android/pull/7163)

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

#### Login with WP.com

- Launch the app
- Log out if needed, and skip onboarding if needed
- Tap "Continue With WordPress.com"
- Enter an invalid WP.com email and tap "Continue" --> an error screen should be shown with a "What is WordPress.com?" link above the "Need help finding the required email?" link. There should be a "Log in with store address" at the bottom above the "Log in with another account" CTA
- Tap on "What is WordPress.com?" CTA --> a webview about WP.com should be shown
- Navigate back and tap on "Need help finding the required email?" CTA --> an alert should be shown as before

#### Login with site address

- Launch the app
- Log out if needed, and skip onboarding if needed
- Tap "Enter Your Store Address"
- Enter an invalid WP.com email and tap "Continue" --> an error screen should be shown with a "What is WordPress.com?" link above the "Need help finding the required email?" link. There should **not** be a "Log in with store address" at the bottom above the "Log in with another account" CTA

### Screenshots
<!-- Include before and after images or gifs when appropriate. -->

\ | light | dark
-- | -- | --
login with WP.com | ![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 11 23 07](https://user-images.githubusercontent.com/1945542/184791624-044d71dc-0daf-4d76-bab8-edc9a8f64f68.png) | ![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 11 23 46](https://user-images.githubusercontent.com/1945542/184791628-919d96fb-4920-4c0e-aa6c-a978764c1ceb.png)
login with store address | ![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 11 24 37](https://user-images.githubusercontent.com/1945542/184791630-6b473e84-0a64-45c2-a640-5872dd89241b.png) | ![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 11 24 45](https://user-images.githubusercontent.com/1945542/184791631-a660fc13-0bac-4bda-9952-d5ae90c3ef73.png)


---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

jaclync committed:
Add an auxiliary view to the bottom of the stack view in `ULErrorViewController` for the new "What is WordPress.com" CTA.

jaclync committed:
Update copy on the invalid email screen.

jaclync committed:
Update release notes.

selanthiraiyan reviewed (commented):
Thanks for working on this, @jaclync 

<img src="https://user-images.githubusercontent.com/524475/184806015-ecba2519-4e8a-4602-9837-0900a5369bd7.jpeg" width="300"/>
1. `Log in with another account` being the only button in the screen should we style it as a primary button? 

----

<img src="https://user-images.githubusercontent.com/524475/184806119-31d5b3ad-64a9-4f6b-a722-b252ad76d7bf.png" width="300"/>
2. The views overlap in small screen size (iPod touch). Do you think it makes sense to introduce scroll behaviour in the stack view?

jaclync committed:
ULErrorViewController: add a scroll view for its dynamic content.

jaclync committed:
Show the only CTA as the primary CTA for login with site address invalid email error.

jaclync committed:
Update buttons from `NUXButton` to `UIButton` to support dynamic type.

jaclync committed:
NotWPAccountViewModel: update unit tests based on the button style changes.

jaclync commented:
Thanks for the suggestions @selanthiraiyan! Btw, you can put the screenshot URLs in an `img` tag like `<img src="url" width="300" />` so that the screenshots aren't shown as very large by default 🙂 

> Log in with another account being the only button in the screen should we style it as a primary button?

Good point, updated in https://github.com/woocommerce/woocommerce-ios/pull/7485/commits/26e0ebbc14a67c3c075bc5658f976a0aa0b88d1d

> The views overlap in small screen size (iPod touch). Do you think it makes sense to introduce scroll behaviour in the stack view?

Nice catch, using a scroll view also helps with dynamic type https://github.com/woocommerce/woocommerce-ios/pull/7485/commits/24c5fdaf4a2b9900c25545936e5c315f0a3c857d. I had to update the constraints a bit to use a top constraint for the content view now that it's in a scroll view. This change also addresses the issue where the text is often truncated in larger fonts in other error screens before this PR. I also updated the bottom CTAs from `NUXButton` to `UIButton` to support dynamic type:

large font | small font
-- | --
![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 17 15 50](https://user-images.githubusercontent.com/1945542/185015164-be48e403-7cde-4d28-ad86-9811698da9d9.png) | ![Simulator Screen Shot - iPhone 13 - 2022-08-16 at 17 16 13](https://user-images.githubusercontent.com/1945542/185015171-fc02f375-f3d0-449d-bd75-c134b6854af4.png)

selanthiraiyan reviewed (approved):
Thank you for making the error screen better than before, Jaclyn. 🙇 

LGTM :shipit:

jaclync committed:
Merge branch 'trunk' into feat/7483-invalid-wpcom-email-error
